syntax = "proto3";

package calculator;


// Define the request message with two numbers
message CalculationRequest {
    int32 num1 = 1;
    int32 num2 = 2;
}

message CalculationResponse {
    int32 result = 1;
}

message LogEntry {
    string operation = 1;
    int32 result = 2;
}

message LogRequest {}

service AdditionService {
    rpc Add(CalculationRequest) returns (CalculationResponse);
    rpc StreamLogs(LogRequest) returns (stream LogEntry); // Streaming logs
}

service MultiplicationService {
    rpc Multiply(CalculationRequest) returns (CalculationResponse);
    rpc StreamLogs(LogRequest) returns (stream LogEntry); // Streaming logs
}
